### PR TITLE
Truncate email addresses that are more than 244 characters long

### PR DESCRIPTION
**Pull request by StCyr**
Fixes https://github.com/nextcloud/mail/issues/3608

Signed-off-by: Cyrille Bollu <cyr.debian@bollu.be>

### CODE REVIEW

**Review by miaulalala — COMMENTED** [review]
Code is fine - but what's the consequences for replying in the in the UI?

**Comment by StCyr** [comment]
> 
> 
> Code is fine - but what's the consequences for replying in the in the UI?

Yes, that was one of my concerns: Replying just won't work.

I've added the "<truncated>" substring to make it more clear to the user that the email address is not usable anymore. 

But, let's be honest, emails coming from such addresses are most probably spam

**Comment by miaulalala** [comment]
> > Code is fine - but what's the consequences for replying in the in the UI?
> 
> Yes, that was one of my concerns: Replying just won't work.
> 
> I've added the "" substring to make it more clear to the user that the email address is not usable anymore.
> 
> But, let's be honest, emails coming from such addresses are most probably spam

True. Is the "" substring clearly identifiable by the user?

**Comment by StCyr** [comment]
> True. Is the "" substring clearly identifiable by the user?

(fun "<truncated>" ("< t r u n c a t ed >") seems to have been truncated by Github)

Not much indeed but we could use it to show it differently in the frontend (in red? with an exclamation mark?)

**Comment by miaulalala** [comment]
The time for a migration would be now.

I like the idea of having an error state for mails, this could be very useful.

**Comment by ChristophWurst** [comment]
- [ ] Blocked by https://github.com/nextcloud/mail/issues/4735

File: lib/Db/MessageMapper.php [hunk 1]
```diff
@@ -341,7 +341,7 @@ public function insertBulk(Account $account, Message ...$messages): void {
 						$qb2->setParameter('message_id', $messageId, IQueryBuilder::PARAM_INT);
 						$qb2->setParameter('type', $type, IQueryBuilder::PARAM_INT);
 						$qb2->setParameter('label', mb_strcut($recipient->getLabel(), 0, 255), IQueryBuilder::PARAM_STR);
-						$qb2->setParameter('email', $recipient->getEmail(), IQueryBuilder::PARAM_STR);
+						$qb2->setParameter('email', mb_substr($recipient->getEmail(),0 , 255), IQueryBuilder::PARAM_STR);
```
Comment (ChristophWurst) [inline lib/Db/MessageMapper.php:344]: adjusted to drop the `<truncated>` because it would be tricky with translations

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**